metric/internal/global/instruments_test.go

```diff
@@ -0,0 +1,329 @@
+				name: "testName",
+				opts: []instrument.Option{},
+			}
+			finish := make(chan struct{})
+			go func() {
+				for {
+					delegate.Add(context.Background(), 1)
+					select {
+					case <-finish:
+						return
+					default:
+					}
+				}
+			}()
+
+			delegate.setDelegate(nonrecording.NewNoopMeter())
+			close(finish)
+		})
+
+		t.Run("Sync Histogram", func(t *testing.T) {
+			delegate := &sfHistogram{
+				name: "testName",
+				opts: []instrument.Option{},
+			}
+			finish := make(chan struct{})
+			go func() {
+				for {
+					delegate.Record(context.Background(), 1)
+					select {
+					case <-finish:
+						return
+					default:
+					}
+				}
+			}()
+
+			delegate.setDelegate(nonrecording.NewNoopMeter())
+			close(finish)
+		})
+	})
+
+	// Int64 Instruments
+
+	t.Run("Int64 Instruments", func(t *testing.T) {
+		t.Run("Sync Counter", func(t *testing.T) {
+			delegate := &siCounter{
+				name: "testName",
+				opts: []instrument.Option{},
+			}
+			finish := make(chan struct{})
+			go func() {
+				for {
+					delegate.Add(context.Background(), 1)
+					select {
+					case <-finish:
+						return
+					default:
+					}
+				}
+			}()
+
+			delegate.setDelegate(nonrecording.NewNoopMeter())
+			close(finish)
+		})
+
+		t.Run("Sync UpDownCounter", func(t *testing.T) {
+			delegate := &siUpDownCounter{
+				name: "testName",
+				opts: []instrument.Option{},
+			}
+			finish := make(chan struct{})
+			go func() {
+				for {
+					delegate.Add(context.Background(), 1)
+					select {
+					case <-finish:
+						return
+					default:
+					}
+				}
+			}()
+
+			delegate.setDelegate(nonrecording.NewNoopMeter())
+			close(finish)
+		})
+
+		t.Run("Sync Histogram", func(t *testing.T) {
+			delegate := &siHistogram{
+				name: "testName",
+				opts: []instrument.Option{},
+			}
+			finish := make(chan struct{})
+			go func() {
+				for {
+					delegate.Record(context.Background(), 1)
+					select {
+					case <-finish:
+						return
+					default:
+					}
+				}
+			}()
+
+			delegate.setDelegate(nonrecording.NewNoopMeter())
+			close(finish)
+		})
+	})
+}
+
+type testCountingFloatInstrument struct {
+	count int
+
+	instrument.Asynchronous
+	instrument.Synchronous
+}
+
+func (i *testCountingFloatInstrument) Observe(context.Context, float64, ...attribute.KeyValue) {
+	i.count++
+}
+func (i *testCountingFloatInstrument) Add(context.Context, float64, ...attribute.KeyValue) {
+	i.count++
+}
+func (i *testCountingFloatInstrument) Record(context.Context, float64, ...attribute.KeyValue) {
+	i.count++
+}
+
+type testCountingIntInstrument struct {
+	count int
+
+	instrument.Asynchronous
+	instrument.Synchronous
+}
+
+func (i *testCountingIntInstrument) Observe(context.Context, int64, ...attribute.KeyValue) {
+	i.count++
+}
+func (i *testCountingIntInstrument) Add(context.Context, int64, ...attribute.KeyValue) {
+	i.count++
+}
+func (i *testCountingIntInstrument) Record(context.Context, int64, ...attribute.KeyValue) {
+	i.count++
+}
```

Comment:
If anything I think it would be instruments_types_test.go, or the top of the file.  Is this is a strong preference?

Reply:
Not a strong preference. I was thinking `meter_types_test.go` because it already contained other types, but, honestly, this is a nit pick. I fine with whatever you decide (including keeping them where they are).